pkg/controller/pravegacluster/upgrade.go

```diff
@@ -444,12 +444,29 @@ func (r *ReconcilePravegaCluster) syncSegmentStoreVersionTo07(p *pravegav1alpha1
 	//this check is to see if the newsts is present or not if it's not present it will be created here
 	if err != nil {
 		if errors.IsNotFound(err) {
+			if p.Spec.ExternalAccess.Enabled {
```

Comment:
It would be good to create the services after the new STS is created. Just move this if block  to ln 469

Reply:
Encapsulate all the code inside the if block in a method and invoke the same here, instead of adding code inline.

Reply:
> It would be good to create the services after the new STS is created. Just move this if block to ln 469

We wont be able to do that because sts is created with "0" replicas and in pravega.spec.replicas it will set the value., due to that external services wont get created. And we are creating sts in case of external service creation fails.

Reply:
> Encapsulate all the code inside the if block in a method and invoke the same here, instead of adding code inline.

Done